base/abstractarray.jl

```diff
@@ -1829,19 +1829,24 @@ function mapslices(f, A::AbstractArray, dims::AbstractVector)
     R[ridx...] = r1
 
     nidx = length(otherdims)
+    indexes = Iterators.drop(CartesianRange(itershape), 1)
+    inner_mapslices!(safe_for_reuse, indexes, nidx, idx, otherdims, ridx, Aslice, A, f, R)
+end
+
+@noinline function inner_mapslices!(safe_for_reuse, indexes, nidx, idx, otherdims, ridx, Aslice, A, f, R)
     if safe_for_reuse
```

Comment:
Ok, well, I did refactor a bit to avoid code duplication but not quite in the way you suggested.